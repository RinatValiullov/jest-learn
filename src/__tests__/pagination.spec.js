import { shallow } from "enzyme";
import React from "react";
import Pagination from "../Pagination/Pagination";

const setupComponent = (props) =>
  shallow(<Pagination {...props} lastPage={20} />);

describe("Pagination component", () => {
  it("should render Pagination without props", () => {
    const component = shallow(<Pagination />);
    expect(component).toMatchSnapshot();
  });

  it("should render Pagination with props", () => {
    const component = setupComponent();
    expect(component).toMatchSnapshot();
  });

  it("should render Pagination for last pages", () => {
    const component = setupComponent({ page: 15 });
    expect(component).toMatchSnapshot();
  });

  it("should render Pagination without dots in the middle", () => {
    const component = setupComponent({ page: 16 });
    expect(component).toMatchSnapshot();
  });

  it("should render Pagination with dots and 3 buttons in the end", () => {
    const component = setupComponent({ page: 19 });
    expect(component).toMatchSnapshot();
  });

  describe("defaultProps", () => {
    it("should use default onClick", () => {
      const result = Pagination.defaultProps.onClick();
      expect(result).toBe(undefined);
    });
  });
});
