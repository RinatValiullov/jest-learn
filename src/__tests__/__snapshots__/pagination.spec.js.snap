// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render Pagination for last pages 1`] = `
<div
  className="paginationWrapper"
>
  <button
    data-name="prev"
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
  <button
    className=""
    data-name={13}
    key="13"
    onClick={[Function]}
  >
    13
  </button>
  <button
    className=""
    data-name={14}
    key="14"
    onClick={[Function]}
  >
    14
  </button>
  <button
    className="active"
    data-name={15}
    key="15"
    onClick={[Function]}
  >
    15
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;

exports[`Pagination component should render Pagination with dots and 3 buttons in the end 1`] = `
<div
  className="paginationWrapper"
>
  <button
    data-name="prev"
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className="active"
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
</div>
`;

exports[`Pagination component should render Pagination with props 1`] = `
<div
  className="paginationWrapper"
>
  <button
    className="active"
    data-name={0}
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className=""
    data-name={1}
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className=""
    data-name={2}
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;

exports[`Pagination component should render Pagination without dots in the middle 1`] = `
<div
  className="paginationWrapper"
>
  <button
    data-name="prev"
    onClick={[Function]}
  >
    &lt;&lt;
  </button>
  <button
    className=""
    data-name={14}
    key="14"
    onClick={[Function]}
  >
    14
  </button>
  <button
    className=""
    data-name={15}
    key="15"
    onClick={[Function]}
  >
    15
  </button>
  <button
    className="active"
    data-name={16}
    key="16"
    onClick={[Function]}
  >
    16
  </button>
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;

exports[`Pagination component should render Pagination without props 1`] = `
<div
  className="paginationWrapper"
>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;
